crates/rooch-framework/sources/auth_validator/schnorr_validator.move

```diff
@@ -14,15 +22,83 @@ module rooch_framework::schnorr_validator {
       SCHEME_SCHNORR
    }
 
+   public fun schnorr_hash(payload: &vector<u8>): u8 {
+      let hash = vector::empty<u8>();
+      let i = SCHNORR_SCHEME_LENGTH + SCHNORR_SIG_LENGTH + SCHNORR_PUBKEY_LENGTH;
+      while (i < SCHNORR_SCHEME_LENGTH + SCHNORR_SIG_LENGTH + SCHNORR_PUBKEY_LENGTH + SCHNORR_HASH_LENGTH) {
+         let value = vector::borrow(payload, i);
+         vector::push_back(&mut hash, *value);
+         i = i + 1;
+      };
+      let vector_size: u64 = vector::length(&hash);
+      let hash_value: u8 = *vector::borrow(&hash, vector_size - 1);
+
+      hash_value
+   }
+
+   public fun schnorr_public_key(payload: &vector<u8>): vector<u8> {
+      let public_key = vector::empty<u8>();
+      let i = SCHNORR_SCHEME_LENGTH + SCHNORR_SIG_LENGTH;
+      while (i < SCHNORR_SCHEME_LENGTH + SCHNORR_SIG_LENGTH + SCHNORR_PUBKEY_LENGTH) {
+         let value = vector::borrow(payload, i);
+         vector::push_back(&mut public_key, *value);
+         i = i + 1;
+      };
+
+      public_key
+   }
+
+   public fun schnorr_signature(payload: &vector<u8>): vector<u8> {
+      let sign = vector::empty<u8>();
+      let i = SCHNORR_SCHEME_LENGTH;
+      while (i < SCHNORR_SIG_LENGTH + 1) {
+         let value = vector::borrow(payload, i);
+         vector::push_back(&mut sign, *value);
+         i = i + 1;
+      };
+
+      sign
+   }
+
+   /// Get the authentication key of the given authenticator.
+   public fun schnorr_authentication_key(payload: &vector<u8>): vector<u8> {
+      let public_key = schnorr_public_key(payload);
+      let addr = schnorr_public_key_to_address(public_key);
+      moveos_std::bcs::to_bytes(&addr)
+   }
+
+   public fun schnorr_public_key_to_address(public_key: vector<u8>): address {
+      let bytes = vector::singleton((SCHEME_SCHNORR as u8));
+      vector::append(&mut bytes, public_key);
+      moveos_std::bcs::to_address(hash::blake2b256(&bytes))
+   }
+
+   public fun get_authentication_key(ctx: &StorageContext, addr: address): vector<u8> {
+      let auth_key_option = account_authentication::get_authentication_key<SchnorrValidator>(ctx, addr);
+      if(option::is_some(&auth_key_option)){
+         option::extract(&mut auth_key_option)
+      }else{
+        //if AuthenticationKey does not exist, return addr as authentication key
+        moveos_std::bcs::to_bytes(&addr)
```

Comment:
The address is not a Shnorr public key's hash, so we can not use the address as authentication_key.

Reply:
Yes. May need to support other non-ed25519's auth keys as built-in schemes.

Reply:
non-ed25519's auth keys need to be inited first, then the validator uses the auth keys.